jerry-port/default/default-promise.c

```diff
@@ -0,0 +1,42 @@
+/* Copyright JS Foundation and other contributors, http://js.foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "jerryscript-port.h"
+#include "jerryscript-port-default.h"
+
+/**
+ * Default implementation of jerry_port_report_unhandled_promise_rejection.
+ * Prints the reason of the unhandled rejections.
+ *
+ * This function called when:
+ *  - A promise is rejected without any handlers
+ *  - A handler is added to a rejected promise for the first time
+ */
+void
+jerry_port_report_unhandled_promise_rejection (const jerry_value_t promise, /**< rejected promise */
+                                               const jerry_value_t reason) /**< reason of the rejection */
+{
+  (void) promise;
+
+  jerry_value_t reason_to_string = jerry_value_to_string (reason);
+  jerry_size_t req_sz = jerry_get_utf8_string_size (reason_to_string);
+  JERRY_VLA (jerry_char_t, str_buf_p, req_sz + 1);
+  jerry_string_to_utf8_char_buffer (reason_to_string, str_buf_p, req_sz);
+  str_buf_p[req_sz] = '\0';
+
+  jerry_release_value (reason_to_string);
+
+  jerry_port_log (jerry_port_default_get_log_level (), "Uncaught (in promise) %s\n", str_buf_p);
```

Comment:
Why `jerry_port_default_get_log_level ()`? Looks like a hack to make the log always pass through. Why not a fixed WARNING level?

Reply:
Warning will be fine.